platform-operator/platformcontroller.go

```diff
@@ -344,6 +346,58 @@ func (c *Controller) deleteFoo(obj interface{}) {
 	c.recorder.Event(foo, corev1.EventTypeNormal, SuccessSynced, MessageResourceSynced)
 }
 
+func (c *Controller) updateFoo(oldObj, newObj interface{}) {
+	fmt.Println("Inside update Foo")
+
+	var err error
+	if _, err = cache.MetaNamespaceKeyFunc(newObj); err != nil {
+		panic(err)
+	}
+
+	oldFoo := oldObj.(*platformworkflowv1alpha1.ResourceComposition)
+	newFoo := newObj.(*platformworkflowv1alpha1.ResourceComposition)
+
+	fmt.Printf("JKL - update\n")
+	fmt.Printf("old: %v\n", oldFoo.Spec)
+	fmt.Printf("new: %v\n", newFoo.Spec)
+
+	namespace := newFoo.ObjectMeta.Namespace
+	if namespace == "" {
+		namespace = "default"
+	}
+	fmt.Printf("GHI - update\n")
+	fmt.Printf("NS:%s", namespace)
+	kind      := newFoo.Spec.NewResource.Resource.Kind
+	group     := newFoo.Spec.NewResource.Resource.Group
+	version   := newFoo.Spec.NewResource.Resource.Version
+	plural    := newFoo.Spec.NewResource.Resource.Plural
+	chartURL  := newFoo.Spec.NewResource.ChartURL
+	chartName := newFoo.Spec.NewResource.ChartName
+	fmt.Printf("Kind:%s, Version:%s Group:%s, Plural:%s\n", kind, version, group, plural)
+	fmt.Printf("ChartURL:%s, ChartName:%s\n", chartURL, chartName)
+
+	// only update if spec has changed
+	if !reflect.DeepEqual(oldFoo.Spec, newFoo.Spec) {
```

Comment:
Interesting. So here we are comparing not only chartURL but the entire spec.

Reply:
The reason I added this is because `updateFunc` will be called even when an update isn't explicitly made. For example, creating a new CRD instance will call `AddFunc`, and afterwards it will call `updateFunc`. Since the update call isn't necessary in this case, we don't want to waste time redownloading and untaring the chart. This was a simple way to make sure to only update when something in the spec has actually changed.

Reply:
I see. Makes sense.